microcosm_sagemaker/testing/directory_comparison.py

```
@@ -19,6 +19,7 @@ def directory_comparison(
     gold_dir: Path,
     actual_dir: Path,
     matchers: Optional[Mapping[Path, ExtractorMatcherPair]] = None,
+    ignore_hidden: bool = True,
 ):
     """
     Recursively checks the contents of `actual_dir` against the expected
```

My PR had added a check that the input directories exist because if not, then the test actually silently passes. This is bad because someone could get the paths wrong and think their test passes. Would be worth adding a test for this corner case (pass in nonexistent paths)

---

Good catch! I added `test_nonexistent_directory_comparison` to `test_directory_comparison.py`.